add log out button

AntonioLok opened the pull request:
# Description

This PR implements the follow:
1 - When user is authenticated, the log out button is displayed. If a token has expired, the token is removed from the local storage.

# Test

### 1- Set-up/run the server and web app  first
#### Backend
Set-up backend server and start the server
0 - Checkout master branch `git checkout master`.
1 - Install latest dependencies `npm i`.
2 - Set the required environment variable `JWT_SECRET` by creating a `.env` file in the main directory and set your secret in that file ie. `JWT_SECRET=HELLO`.
3 - run the app using `npm run start`. If you want to skip step 2, you can set the env with `JWT_SECRET=HELLO npm run start`.

#### Front-end
Set-up frontend webapp and start the app:
0 - Checkout this feature branch `git checkout add-log-out-button`.
1 - Install latest dependencies `npm i`.
2 - run the app using `npm run start`.

### 2- After the server and web app have been started, navigate to the login page by clicking on the login button. Login with your credentials that you used last time or use the following one:
```
email: test@gmail.com
password: test@gmail.com
```

Upon logging in, the `logout` button should now appear. 

### 3 - To check that expired tokens are removed, first log out by clicking on the "Log out" button. Then on the backend repository, go to `src/controllers/user/generate-token.js`. On line 25, change `{ expiresIn: 86400 }` to `{ expiresIn: 1 }`. And restart the server. 

### 4 - Repeat #2 by re-logging in and wait 1s. Then on the console, type `localStorage.getItem('OS_AUTH_TOKEN');` to retrieve the token. You should receive null since the token has been removed. Similarly if you refresh the website, the "Log in" button should appear.

AntonioLok commented:
I know you have been very busy but just in case you forgot about this PR... :). @howardhenry

howardhenry commented on src/components/app.js:
```diff
@@ -17,20 +21,56 @@ const {
+class App extends Component {
+  shouldComponentUpdate(nextProps) {
+    const { loginState } = this.props;
+    // deep comparisons
```
You can remove this comment as the function name should be descriptive enough.

howardhenry commented on src/components/app.js:
```diff
@@ -17,20 +21,56 @@ const {
+      } else {
+        localStorage.removeItem('OS_AUTH_TOKEN');
+      }
+    }
```
The logic to check if the use is authenticated doesn't belong inside a component. This should be somewhere like an authentication helper to ensure separation of concerns and reusability.

howardhenry commented on src/components/common/nav-menu/index.js:
```diff
@@ -1,13 +1,32 @@
+  logOut() {
+    localStorage.removeItem('OS_AUTH_TOKEN');
+  }
+
```
Like the authentication check, the log out function should not be defined in a component. Move this to an authentication helper/util as you will likely reuse the function again in a different component or module.

AntonioLok commented:
> So sorry I completely missed this one!
> 
> It looks great! I've just added a few comments to keep up best practices.

No worries. Thanks for the feedback. I have updated the PR as per comments. Thanks for your time!
